File: snippets/fsharp/System/Type/GetTypeFromCLSID/gettypefromclsid_ex2.fs
module gettypefromclsid_ex2

// <Snippet2>
open System
open System.Reflection

let [<Literal>] WORD_CLSID = "{000209FF-0000-0000-C000-000000000046}"

try
    // Start an instance of the Word application.
    let word = Type.GetTypeFromCLSID(Guid.Parse WORD_CLSID, true)
    printfn $"Instantiated Type object from CLSID {WORD_CLSID}"
    let wordObj = Activator.CreateInstance word
    printfn $"Instantiated {wordObj.GetType().FullName} from CLSID {WORD_CLSID}" 
    
    // Close Word.
    word.InvokeMember("Quit", BindingFlags.InvokeMethod, null, wordObj, [| box 0; 0; false |] ) |> ignore
with _ ->
    printfn $"Unable to instantiate an object for {WORD_CLSID}"
// The example displays the following output:
//    Instantiated Type object from CLSID {000209FF-0000-0000-C000-000000000046}
//    Instantiated Microsoft.Office.Interop.Word.ApplicationClass
// </Snippet2>

File: snippets/csharp/System/Type/GetTypeFromCLSID/gettypefromclsid_ex2.cs
// <Snippet2>
using System;
using System.Reflection;


public class GetTypeFromClsidExample2
{
    private const string WORD_CLSID = "{000209FF-0000-0000-C000-000000000046}";

    public static void Run()
    {
        try
        {
            // Start an instance of the Word application.
            var word = Type.GetTypeFromCLSID(Guid.Parse(WORD_CLSID), true);
            Console.WriteLine("Instantiated Type object from CLSID {0}",
                              WORD_CLSID);
            object wordObj = Activator.CreateInstance(word);
            Console.WriteLine("Instantiated {0}",
                              wordObj.GetType().FullName, WORD_CLSID);

            // Close Word.
            word.InvokeMember("Quit", BindingFlags.InvokeMethod, null,
                              wordObj, [0, 0, false]);
        }
        catch (Exception)
        {
            Console.WriteLine("Unable to instantiate an object for {0}", WORD_CLSID);
        }
    }
}
// The example displays the following output:
//    Instantiated Type object from CLSID {000209FF-0000-0000-C000-000000000046}
//    Instantiated Microsoft.Office.Interop.Word.ApplicationClass
// </Snippet2>
